rs/xnet/payload_builder/src/lib.rs

```diff
@@ -1612,21 +1579,17 @@ impl XNetClient for XNetClientImpl {
                 Instant::now().saturating_duration_since(request_start),
             );
 
-            let response = result.map_err(|e| {
-                if e.is_timeout() {
-                    XNetClientError::Timeout
-                } else {
-                    XNetClientError::RequestFailed(e)
-                }
-            })?;
+            let response = result.map_err(XNetClientError::RequestFailed)?;
```

Comment:
nit: you can directly do 

`let response = self.http_client.get(endpoint.url.clone()).await.map_err(XNetClientError::RequestFailed)?;`

few lines above